tests/checks/path.fish

```diff
@@ -0,0 +1,210 @@
+#RUN: %fish %s
+# The "path" builtin for dealing with paths
+
+# Extension - for figuring out the file extension of a given path.
+path extension /
+or echo None
+# CHECK: None
+
+# No extension
+path extension /.
+or echo Filename is just a dot, no extension
+# CHECK: Filename is just a dot, no extension
+
+# No extension - ".foo" is the filename
+path extension /.foo
+or echo None again
+# CHECK: None again
+
+path extension /foo
+or echo None once more
+# CHECK: None once more
+path extension /foo.txt
+and echo Success
+# CHECK: txt
+# CHECK: Success
+path extension /foo.txt/bar
+or echo Not even here
+# CHECK: Not even here
+path extension . ..
+or echo No extension
+# CHECK: No extension
+path extension ./foo.mp4
+# CHECK: mp4
+path extension ../banana
+# nothing, status 1
+echo $status
+# CHECK: 1
+path extension ~/.config
+# nothing, status 1
+echo $status
+# CHECK: 1
+path extension ~/.config.d
+# CHECK: d
+path extension ~/.config.
+echo $status
+# one empty line, status 0
+# CHECK: 
+# CHECK: 0
+
+path strip-extension ./foo.mp4
+# CHECK: ./foo
+path strip-extension ../banana
+# CHECK: ../banana
+# but status 1, because there was no extension.
+echo $status
+# CHECK: 1
+path strip-extension ~/.config
+# CHECK: {{.*}}/.config
+echo $status
+# CHECK: 1
+
+path base ./foo.mp4
+# CHECK: foo.mp4
+path base ../banana
+# CHECK: banana
+path base /usr/bin/
+# CHECK: bin
+path dir ./foo.mp4
+# CHECK: .
+path base ../banana
+# CHECK: banana
+path base /usr/bin/
+# CHECK: bin
+
+cd $TMPDIR
```

Comment:
>I've been thinking of adding explicit "this test needs to be run with the test
driver" bits to tests so we can give better error messages when someone
tries to run them directly with littlecheck. Not sure how though.

Tbh... just run `ninja test_path.fish`. That means there is *very little* reason to run the tests otherwise.